include/picolibrary/vector.h

```diff
@@ -0,0 +1,1326 @@
+        return {};
+    }
+
+    /**
+     * \brief Replace the vector's contents with those in the provided initializer list.
+     *
+     * \param[in] initializer_list The initializer list containing the values to store in
+     *            the vector.
+     *
+     * \return Nothing if the size of the initializer list is less than or equal to N.
+     * \return picolibrary::Generic_Error::INSUFFICIENT_CAPACITY if the size of the
+     *         initializer list is greater than N.
+     */
+    constexpr auto assign( std::initializer_list<Value> initializer_list ) noexcept
+        -> Result<Void, Error_Code>
+    {
+        if ( initializer_list.size() > N ) {} // if
+
+        clear();
+
+        ::picolibrary::for_each(
+            initializer_list.begin(), initializer_list.end(), [ storage = &m_storage[ 0 ] ]( auto value ) noexcept {
+                new ( storage++ ) Value{ value };
+            } );
+
+        return {};
+    }
+
+    /**
+     * \brief Access the element at the specified position in the vector.
+     *
+     * \warning Bounds checking is not performed.
+     *
+     * \param[in] position The position of the vector element to access.
+     *
+     * \return The element at the specified position in the vector.
+     */
+    auto operator[]( Position position ) noexcept -> Reference
+    {
+        return data()[ position ];
+    }
+
+    /**
+     * \brief Access the element at the specified position in the vector.
+     *
+     * \warning Bounds checking is not performed.
+     *
+     * \param[in] position The position of the vector element to access.
+     *
+     * \return The element at the specified position in the vector.
+     */
+    auto operator[]( Position position ) const noexcept -> Const_Reference
+    {
+        return data()[ position ];
+    }
+
+    /**
+     * \brief Access the element at the specified position in the vector.
+     *
+     * \param[in] position The position of the vector element to access.
+     *
+     * \return The element at the specified position in the vector if the specified
+     *         position is a valid position.
+     * \return picolibrary::Generic_Error::OUT_OF_RANGE if the specified position is not a
+     *         valid position.
+     */
+    auto at( Position position ) noexcept -> Result<std::reference_wrapper<Value>, Error_Code>
+    {
+        if ( position >= size() ) {
+            return Generic_Error::OUT_OF_RANGE;
+        } // if
+
+        return data()[ position ];
+    }
+
+    /**
+     * \brief Access the element at the specified position in the vector.
+     *
+     * \param[in] position The position of the vector element to access.
+     *
+     * \return The element at the specified position in the vector if the specified
+     *         position is a valid position.
+     * \return picolibrary::Generic_Error::OUT_OF_RANGE if the specified position is not a
+     *         valid position.
+     */
+    auto at( Position position ) const noexcept
```

Comment:
This function can be `constexpr` since it has an execution path that doesn't use `reinterpret_cast`.
```suggestion
    constexpr auto at( Position position ) const noexcept
```